dotCMS/src/integration-test/java/com/dotcms/enterprise/rules/RulesAPIImplIntegrationTest.java

```diff
@@ -527,6 +702,30 @@ public void shouldSaveNewRuleInPage() throws DotSecurityException, DotDataExcept
         assertTrue(rulesId.contains(rule.getId()));
     }
 
+    @NotNull
+    private void addPermission(
+            final Role role,
+            final Permissionable permissionable,
+            final int permissionPublish) {
+
+        final Permission publishPermission = getPermission(role, permissionable, permissionPublish);
+
+        try {
+            APILocator.getPermissionAPI().save(publishPermission, permissionable, systemUser, false);
+        } catch (DotDataException | DotSecurityException e){
+            throw new RuntimeException(e);
+        }
+    }
+
+    @NotNull
+    private Permission getPermission(Role role, Permissionable permissionable, int permissionPublish) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'role' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5259424)